python/fledge/common/configuration_manager.py

```diff
@@ -516,8 +516,12 @@ async def update_configuration_item_bulk(self, category_name, config_item_list):
                         raise TypeError('Unrecognized value name for item_name {}'.format(item_name))
 
                 if 'mandatory' in cat_info[item_name]:
-                    if cat_info[item_name]['mandatory'] == 'true' and not len(new_val.strip()):
-                        raise ValueError("A value must be given for {}".format(item_name))
+                    if cat_info[item_name]['mandatory'] == 'true':
+                        if cat_info[item_name]['type'] == 'JSON':
+                            if not len(new_val):
+                                raise ValueError("A value must be given for {}".format(item_name))
```

Comment:
if new_val supplies with empty dict i.e {}
It will never be caught inside it as the type of new_val is always be in string. In that case length is 2.
You need to convert python str to dict object

Reply:
That may be valid for a JSON type of course. I'm tempted to leave this for now